server/src/controllers/client_timeline_entries/addClientTimelineEntry.js

```diff
@@ -0,0 +1,96 @@
+const logger = require("pino")();
+const ClientTimelineEntry = require("../../models/client_timeline_entry.model");
+const {
+  submitPlacementUpdateEntryInTimelines,
+} = require("../../utils/placement_entry_util");
+
+const addClientTimelineEntryRequestHandler = async (req, res) => {
+  try {
+    // eslint-disable-next-line camelcase
+    const { type, title, body, client, job_lead } = req.body.entry;
+
+    const { user } = req;
+
+    const validTypes = ["update", "contact", "placement", "note"];
+    if (!validTypes.includes(type)) {
+      return res.status(400).json({
+        status: "fail",
+        message: "Type not defined or is not valid",
+        data: null,
+      });
+    }
+
+    if (
+      ["update", "contact"].includes(type) &&
+      (!user || !title || !body || !client)
+    ) {
+      return res.status(400).json({
+        status: "fail",
+        message: `For type ${type}, user, title, body, and client must be defined`,
+        data: null,
+      });
+    }
+
+    if (
+      type === "placement" &&
+      // eslint-disable-next-line camelcase
+      (!user || !title || !body || !client || !job_lead)
+    ) {
+      return res.status(400).json({
+        status: "fail",
+        message:
+          "For type placement, user, title, body, client, and job_lead must be defined",
+        data: null,
+      });
+    }
+
+    if (type === "note" && (!user || !title || !body)) {
+      return res.status(400).json({
+        status: "fail",
+        message: "For type note, user, title, and body must be defined",
+        data: null,
+      });
+    }
+
+    let clientTimelineEntry;
+
+    if (type === "placement") {
+      clientTimelineEntry = submitPlacementUpdateEntryInTimelines(
+        // eslint-disable-next-line camelcase
+        { type, title, body, client, job_lead, user },
+        "client",
+      );
+    } else {
+      clientTimelineEntry = await ClientTimelineEntry.create({
+        date_added: new Date(),
+        type,
+        title,
+        body,
+        client,
+        // eslint-disable-next-line camelcase
+        job_lead,
+        user,
```

Comment:
This should be the user id rather than the user object

Reply:
user variable is the user id

Reply:
<img width="766" alt="Screenshot 2024-03-03 at 4 10 51 PM" src="https://github.com/uoftblueprint/tces/assets/58835213/d945ddde-8cce-430d-971b-1e5a8b6938a9">
Since we access user from the request it is a user object here

Reply:
sounds good i have swtiched it out.